custom_components/plugwise/binary_sensor.py

```diff
@@ -93,36 +96,33 @@ async def async_setup_entry(
     entry: PlugwiseConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up the Plugwise binary_sensors from a ConfigEntry."""
+    """Set up the Plugwise binary_sensors from a config entry."""
     coordinator = entry.runtime_data
 
     @callback
     def _add_entities() -> None:
-        """Add Entities during init and runtime."""
+        """Add Entities."""
         if not coordinator.new_devices:
             return
 
-        entities: list[PlugwiseBinarySensorEntity] = []
-        for device_id in coordinator.new_devices:
-            device = coordinator.data.devices[device_id]
-            if not (binary_sensors := device.get(BINARY_SENSORS)):
-                continue
-            for description in PLUGWISE_BINARY_SENSORS:
-                if description.key not in binary_sensors:
-                    continue
-                entities.append(PlugwiseBinarySensorEntity(coordinator, device_id, description))
-                LOGGER.debug(
-                    "Add %s %s binary sensor", device[ATTR_NAME], description.translation_key
+        async_add_entities(
+            PlugwiseBinarySensorEntity(coordinator, device_id, description)
+            for device_id in coordinator.new_devices
+            if (
+                binary_sensors := coordinator.data.devices[device_id].get(
+                    BINARY_SENSORS  # upstream to const
                 )
-
-        async_add_entities(entities)
+            )
+            for description in PLUGWISE_BINARY_SENSORS  # upstream from const
+            if description.key in binary_sensors
+        )
```

Comment:
I kept the code like this on purpose because I wanted to keep the debug-logging.